File: 8kyu/(8 kyu) Name Shuffler/(8 kyu) Name Shuffler.py
# 1 Plain solution
def name_shuffler(str_):
    spaceIndex = str_.index(" ")
    firstName = str_[0:spaceIndex]
    lastName = str_[spaceIndex + 1:]

    return lastName + " " + firstName

# 2 Optimized solution
def name_shuffler(str_):
    name = str_.split(" ")
    return name[1] + " " + name[0]

# 3 Clever solution
def name_shuffler(str_):
    return " ".join(reversed(str_.split(" ")))

# 4 Coding golf
name_shuffler=lambda n:" ".join(n.split()[::-1])

File: 8kyu/(8 kyu) Name Shuffler/(8 kyu) Name Shuffler.rs
// 1 Plain solution
fn name_shuffler(s: &str) -> String {
    let space_index = s.find(' ').unwrap();
    let first_name = &s[..space_index];
    let last_name = &s[space_index + 1..];

    format!("{} {}", last_name, first_name)
}

// 2 Optimized solution
fn name_shuffler_optimized(s: &str) -> String {
    let mut name = s.find(' ').unwrap();
    format!("{} {}", &s[name + 1..], &s[..name])
}

// 3 Clever solution
fn name_shuffler_clever(s: &str) -> String {
    s.split_whitespace().rev().collect::<Vec<&str>>().join(" ")
}